src/sentry/analytics/base.py

```diff
@@ -1,22 +1,24 @@
+from __future__ import annotations
+
 __all__ = ("Analytics",)
 
+import abc
+from typing import Any
 
 from sentry.analytics.event import Event
 from sentry.utils.services import Service
 
 from .event_manager import default_manager
 
 
-class Analytics(Service):
+class Analytics(Service, abc.ABC):  # type: ignore
```

Comment:
Sorry if these questions are all simple/obvious, what does this module do? (`abc.ABC`) I don't see it used in the class.

Reply:
`ABC` stands for "abstract base class". It marks a class as abstract meaning that you should not initialize it and that it does not have to implement all of its parents' abstract methods.
https://docs.python.org/3/library/abc.html

Reply:
Actually, if we're trying to be pure, an abstract class should not be able to have an `__init__()` method. We're doing tricky things with this class were `type` is abstract-ish, meaning that it can either be passed in by parameter or inheritance. I should remove the `ABC`.

Reply:
That makes a bit more sense, thanks